.github/workflows/maven.yaml

```diff
@@ -66,7 +72,37 @@ jobs:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
-      - name: 'Build Kroxylicious maven project'
+      - name: 'Build & unit test Kroxylicious maven project'
+        id: mvn_build
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          GIT_HASH="$(git rev-parse HEAD)"
+          IMAGE_TAG="dev-git-${GIT_HASH}"
+          KROXYLICIOUS_VERSION="$(mvn org.apache.maven.plugins:maven-help-plugin:3.4.0:evaluate -Dexpression=project.version -q -DforceStdout)"
+          KROXYLICIOUS_IMAGE="quay.io/kroxylicious/kroxylicious:${IMAGE_TAG}"
+          echo "KROXYLICIOUS_VERSION=${KROXYLICIOUS_VERSION}" >> "$GITHUB_ENV"
+          # KROXYLICIOUS_IMAGE env var is used by the Operator ITs
+          echo "KROXYLICIOUS_IMAGE=${KROXYLICIOUS_IMAGE}" >> "$GITHUB_ENV"
+          mvn -B verify -DskipITs -Pci -Djapicmp.skip=${REFERENCE_RELEASE_UNPUBLISHED}
+      - name: 'Set up Docker Buildx'
+        uses: docker/setup-buildx-action@v3
+      - name: 'Build tagged Docker image'
+        uses: docker/build-push-action@v6
+        with:
+          file: Dockerfile
+          context: .
+          build-args:
+            KROXYLICIOUS_VERSION="${{ env.KROXYLICIOUS_VERSION }}"
+#          context: nextjs-blog
+          tags: ${{ env.KROXYLICIOUS_IMAGE }}
+          push: false
+          cache-from: type=gha
```

Comment:
Nice idea - we should apply that to the other workflows doing Docker image builds.

Reply:
Once I get it working 😁